charts/rancher-external-dns/v0.0.1/templates/deployment.yaml

```diff
@@ -0,0 +1,238 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels: {{ include "external-dns.labels" . | indent 4 }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+  name: {{ template "external-dns.fullname" . }}
+spec:
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
+    {{- if .Values.podAnnotations }} 
+{{ toYaml .Values.podAnnotations | indent 8}}
+    {{- end }}
+      labels: {{ include "external-dns.labels" . | indent 8 }}
+    spec:
+      {{- if .Values.image.pullSecrets }}
+      imagePullSecrets:
+      {{- range $sec := .Values.image.pullSecrets }}
+        - name: {{$sec | quote }}
+      {{- end }}
+      {{- end }}
+      {{- if .Values.priorityClassName }}
+      priorityClassName: "{{ .Values.priorityClassName }}"
+      {{- end }}
+      containers:
+        - name: {{ template "external-dns.name" . }}
+          image: "{{.Values.image.name}}:{{ .Values.image.tag }}"
```

Comment:
@prachidamle , everything is fine, but the airgap case should add `private image registry` before `image` field. FYI, https://github.com/rancher/system-charts/pull/11, https://github.com/rancher/system-charts/pull/10

1. Add `global.systemDefaultRegistry` to root `values.yaml`
```
global:
  systemDefaultRegistry: ""
```

2. Add template into `templates/_helpers.tpl`
```
{{- define "system_default_registry" -}}
{{- if .Values.global.systemDefaultRegistry -}}
{{- printf "%s/" .Values.global.systemDefaultRegistry -}}
{{- else -}}
{{- "" -}}
{{- end -}}
{{- end -}}
```

3. Add `system_default_registry` into image field
```
image: {{ template "system_default_registry" . }}{{ .Values.image.name}}:{{ .Values.image.tag }}
```

Reply:
@thxCode I have added these changes, please check. I will give this a test.